### PR TITLE
add LoadFromString method to support offline testing of configs.

### PR DESCRIPTION
**Pull request by brianbegygr**

### CODE REVIEW

File: GuaranteedRate.Sextant/Config/IniConfig.cs [hunk 1]
```diff
@@ -132,5 +113,45 @@ public IEncompassConfig GetConfigGroup(string key)
...
+                    }
+                }
+
+                _config = config;
+                return true;
+            }
+            catch (Exception)
+            {
+                    
+                throw;
```
Comment (jeffpsherman) [inline GuaranteedRate.Sextant/Config/IniConfig.cs]: Why catch and throw?
I know you can't log, just not sure what the value of catching here is

File: GuaranteedRate.Sextant/Config/IniConfig.cs [hunk 2]
```diff
@@ -132,5 +113,45 @@ public IEncompassConfig GetConfigGroup(string key)
...
+            }
```
Comment (jeffpsherman) [inline GuaranteedRate.Sextant/Config/IniConfig.cs]: Should this return `false` if there was an exception?

File: GuaranteedRate.Sextant/Config/JsonEncompassConfig.cs [hunk 3]
```diff
@@ -156,5 +157,24 @@ public IEncompassConfig GetConfigGroup(string key)
...
+        {
+            try
+            {
+                _jsonObject = JObject.Parse(configAsString);
+                return true;
+
+            }
+            catch (Exception)
+            {
+                throw;
```
Comment (jeffpsherman) [inline GuaranteedRate.Sextant/Config/JsonEncompassConfig.cs]: Same questions about the empty catch/throw and returning false

File: GuaranteedRate.Sextant/Config/IniConfig.cs (hunk 2 above)
Comment (brianbegygr) [inline GuaranteedRate.Sextant/Config/IniConfig.cs]: I can do that.

File: GuaranteedRate.Sextant/Config/IEncompassConfig.cs [hunk 4]
```diff
@@ -22,6 +22,10 @@ public interface IEncompassConfig
         bool GetValue(string key, bool defaultValue);
         ICollection<string> GetKeys();
         IEncompassConfig GetConfigGroup(string key);
-        
+
+        bool LoadFromString(string configAsString);
```
Comment (jeffpsherman) [inline GuaranteedRate.Sextant/Config/IEncompassConfig.cs]: Would it make more sense to make this method an overload of Init?
```
public bool Init(String configAsString)
```
It seems like you wouldn't call both Init & LoadFromString, making the new function an overload would make that clear

**Comment by jeffpsherman** [comment]
👍

**Pull request merged**